src/test/java/com/karankumar/bookproject/backend/service/PublisherServiceTest.java

```
@@ -0,0 +1,135 @@
+public class PublisherServiceTest {
+
+    private final PublisherService publisherService;
+    private final BookService bookService;
+
+    @Autowired
+    PublisherServiceTest(PublisherService publisherService, BookService bookService){
+        this.publisherService = publisherService;
+        this.bookService = bookService;
+    }
+
+    @BeforeEach
+    public void setUp() {
+        resetServices();
+    }
+
+    void resetServices() {
+        bookService.deleteAll();
+        publisherService.deleteAll();
+    }
+
+
+    @Test
+    @DisplayName("Save a Publisher Properly")
+    void saveValidPublisher() {
+        // given
+        Publisher publisher = new Publisher("Test SavePublisher ");
+
+        // when
+        publisherService.save(publisher);
+
+        // then
+        assertThat(publisherService.findById(publisher.getId())).isNotNull();
+    }
+
+    @Test
+    @DisplayName("throw exception on an attempt to save a null Publisher")
+    void throwExceptionWhenSavingANullPublisher() {
+        assertThatExceptionOfType(NullPointerException.class)
+                .isThrownBy(() -> publisherService.save(null));
+    }
+
+    @Test
+    @DisplayName("throw exception on an attempt to delete a null Publisher")
+    void throwExceptionWhenDeletingANullPublisher() {
+        assertThatExceptionOfType(NullPointerException.class)
+                .isThrownBy(() -> publisherService.delete(null));
+    }
+
+    @Test
+    @Transactional
+    @DisplayName("Delete a Publisher")
+    void deleteExistingPublisher() {
+        // given
+        Publisher publisher = new Publisher("Test DeletePublisher ");
+        publisherService.save(publisher);
+        Long publisherId = publisher.getId();
+
+        // when
+        publisherService.delete(publisher);
+
+        // then
+        assertThatThrownBy(() -> publisherService.findById(publisherId)).isInstanceOf(
+                JpaObjectRetrievalFailureException.class);
+    }
```

Is it not possible to delete a publisher?

---

Yes, we can. 

What I'm trying here to delete the publisher and checking the existance of that publisher after that. It should throw an error .

---

Ah, my bad. I should have looked closer at the assertion. Thanks for clarifying